src/nvim/screen.c

```diff
@@ -4494,6 +4474,83 @@ void screen_adjust_grid(ScreenGrid **grid, int *row_off, int *col_off)
   }
 }
 
+// Get information needed to display the sign in line 'lnum' in window 'wp'.
+// If 'nrcol' is TRUE, the sign is going to be displayed in the number column.
+// Otherwise the sign is going to be displayed in the sign column.
+static void get_sign_display_info(
+    int nrcol,
+    win_T *wp,
+    linenr_T lnum,
+    int row,
+    int startrow,
+    int filler_lines,
+    int filler_todo,
+    int count,
+    int *c_extrap,
+    int *c_finalp,
+    char_u *extra,
+    size_t extra_size,
+    char_u **pp_extra,
+    int *n_extrap,
+    int *char_attrp,
+    int *draw_statep,
+    int *sign_idxp
+)
+{
+  int text_sign;
+
+  // Draw cells with the sign value or blank.
+  *c_extrap = ' ';
+  *c_finalp = NUL;
+  if (nrcol) {
+    *n_extrap = number_width(wp) + 1;
+  } else {
+    *char_attrp = win_hl_attr(wp, HLF_SC);
+    *n_extrap = win_signcol_width(wp);
+  }
+
+  if (row == startrow + filler_lines && filler_todo <= 0) {
+    text_sign = buf_getsigntype(wp->w_buffer, lnum, SIGN_TEXT,
+                                *sign_idxp, count);
+    if (text_sign != 0) {
+      *pp_extra = sign_get_text(text_sign);
+      if (*pp_extra != NULL) {
+        *c_extrap = NUL;
+        *c_finalp = NUL;
+
+        if (nrcol) {
+          snprintf((char *)extra, extra_size, "%-*s ",
+                   number_width(wp), pp_extra);
```

Comment:
You have a new enbug here when fixing the lint error.

```C 
snprintf((char *)extra, extra_size, "%-*s ",
              number_width(wp), *pp_extra);
```

Reply:
Thanks.